### PR TITLE
Versioning clarifications

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/BC2QUyPW/9-versioning-policy-5-4-x-updates

This commit makes official rules which have been almost in place for the last patch/minor releases and should improve transparency for consumers.

- Precise the OMERO/Bio-Formats versioning decoupling point
- Use [semantic versioning](http://semver.org/) for the Bio-Formats API policy. This policy has been effectively in place since Bio-Formats 5.4.0.
- Increase the stringency of the serialization policy
- Add paragraph about the versioning of individual format changes

In terms of release, as per the usage of semver for the API, one expectation is that it should be possible to have more minor releases of Bio-Formats containing API backwards-compatible additions, format reader additions and/or backwards-incompatible or serialization breaking format reader changes.

### CODE REVIEW

File: docs/sphinx/about/index.rst [hunk 1]
```diff
@@ -50,52 +50,60 @@ include:
...
-Bio-Formats is now decoupled from OMERO with its own release schedule rather
-than being updated whenever a new version of :products_plone:`OMERO <omero>`
-is released.
-We expect this to result in more frequent releases to get fixes out to the
-community faster.
-
-See the :doc:`version history <whats-new>` for a list of major changes in
-each release.
+Since Bio-Formats 5.1.3, Bio-Formats is decoupled from OMERO with its own
+release schedule than being updated whenever a new version of
```
Comment (mtbc) [inline docs/sphinx/about/index.rst]: "than" no longer fits without the "rather"

File: docs/sphinx/about/index.rst [hunk 2]
```diff
@@ -50,52 +50,60 @@ include:
...
+:products_plone:`OMERO <omero>` is released.
+This change allows in more frequent releases to get fixes out to the
```
Comment (mtbc) [inline docs/sphinx/about/index.rst]: "allows in" is odd

File: docs/sphinx/about/index.rst [hunk 3]
```diff
@@ -50,52 +50,60 @@ include:
...
+
+For format reader fixes and additions, the policy should read as follows:
+
+- The minor version Y MUST be incremented when a new file format reader is
+  introduced.
+- The minor version Y MUST be incremented when a non backwards-compatible
+  format bug fix is introduced, e.g. a fix that modifies the core metadata
+  of existing files.
+- The patch version Z MUST be incremented if only backwards-compatible format
+  bug fixes are introducted
```
Comment (mtbc) [inline docs/sphinx/about/index.rst]: "introduced", also needs terminal period

File: docs/sphinx/about/index.rst (hunk 2 above)
Comment (hflynn) [inline docs/sphinx/about/index.rst]: "allows for" would be correct

**Comment by dgault** [comment]
This reads fine to me, it all makes sense and should be easily understood by users and developers who would be concerned with breaking changes. Looks good to merge from my perspective.

**Comment by hflynn** [comment]
Looks good to me too

**Pull request merged**